src/app/modules/project/add/manual/manual.component.ts

```diff
@@ -123,20 +132,41 @@ export class ManualComponent implements OnInit {
 
     const newProject: ProjectAdd = this.newProjectForm.value;
     newProject.collaborators = this.collaborators;
+    // Start uploading files
+    this.fileUploader.uploadFiles()
+        .subscribe(uploadedFiles => {
+          if (uploadedFiles) {
+            if (uploadedFiles[0]) {
+              // Project icon was set and uploaded
+              newProject.fileId = uploadedFiles[0].id;
+              this.createProject(newProject);
+            }
+            // Project icon was set but not uploaded successfully, the component will show the error
+          } else {
+            // There was no project icon
+            newProject.fileId = 0;
+            this.createProject(newProject);
+          }
 
+        });
+
+
+  }
+
+  private createProject(newProject): void {
     this.projectService
-      .post(newProject)
-      .pipe(finalize(() => (this.submitEnabled = false)))
-      .subscribe(() => {
-        const alertConfig: AlertConfig = {
-          type: AlertType.success,
-          mainMessage: 'Project was succesfully saved',
-          dismissible: true,
-          timeout: this.alertService.defaultTimeout
-        };
-        this.alertService.pushAlert(alertConfig);
-        this.router.navigate([`/project/overview`]);
-      });
+        .post(newProject)
+        .pipe(finalize(() => (this.submitEnabled = false)))
+        .subscribe(() => {
+          const alertConfig: AlertConfig = {
+            type: AlertType.success,
+            mainMessage: 'Project was succesfully saved',
+            dismissible: true,
+            timeout: this.alertService.defaultTimeout
```

Comment:
You can remove the timeout property from the object when using the default value